proposals/etcd.adoc

```diff
@@ -0,0 +1,104 @@
+== etcd Project Proposal
+
+*Name of Project*: etcd
+
+*Description*:
+
+etcd is a consistent distributed key-value store, designed to hold small
+amounts of data that can fit entirely in memory (although etcd still writes
+to disk for durability) and mainly used as a separate coordination service
+for other distributed systems like Kubernetes. A typical etcd cluster is
+distributed over 3 to 5 nodes, for high availability, while it prioritizes
+consistency and partition tolerance. Which means, it provides one logical
+cluster view of many physical servers. So long as a majority of cluster is up,
+etcd continues to work, even under machine failures. This redundancy provides
+fault tolerance.
+
+etcd server implements Raft consensus algorithm for data replication. Raft is
+a leader-based protocol. Data is replicated from leader to follower; follower
+forwards proposals to leader, and leader decides what to commit or not. Leader
+persists and replicates an entry, once it has been agreed by the quorum of
+cluster. Therefore, adding more members to cluster would lead to more
+communication overhead and limited write scalability. Write (and consistent
+read) throughput is limited to durable write throughput of a single leader
+node.
+
+The underlying storage layer for Raft log is write-ahead log (WAL).
+Committed entries are written out to disk, so they can be replayed on restart.
+etcd uses gRPC for transport layer. Client employs HTTP/2 Ping for server
+health checking, and implements automatic failover under faulty networks.
+
+*Sponsor / Advisor from TOC*: Brian Grant <briangrant@google.com>
+
+*Unique Identifier*: etcd
+
+*License*: https://github.com/coreos/etcd/blob/master/LICENSE[Apache License 2.0]
+
+*Maturity Level*: Incubating
+
+*Source control repositories*: https://github.com/etcd-io (to be moved
+from https://github.com/coreos/etcd/issues/9965[August 27, 2018])
+
+*Issue tracker*: https://github.com/coreos/etcd/issues
+
+*Initial committers*: https://github.com/philips[Brandon Phillips] and
```

Comment:
If the full list of committers/maintainers is in https://github.com/coreos/etcd/blob/master/MAINTAINERS, then I suggest we remove the 2 specific individuals from here. There are active maintainers from at least Redhat (CoreOS), Amazon, and Google, correct?

Reply:
I think there is a misunderstanding of what "Initial Committers" means. Does it mean the people who were initially on the project when it started or the people who will be on the project when donated to CNCF?

For context Xiang and I started the project 5 years ago so I assume that is why we are listed here.

Reply:
@bgrant0607 @philips I took this from https://github.com/cncf/toc/blob/master/proposals/grpc.adoc, where initial "git" committers are listed as "Initial committers" (ref. https://github.com/grpc/grpc-java/graphs/contributors?from=2014-05-04&to=2014-09-30&type=c).

Reply:
Sorry, this may not be all that important, but perhaps @caniszczyk has guidance.

Reply:
@caniszczyk Shall we remove?

Reply:
@gyuho this means who has commit access to the project when donated, essentially we want to see who has write access, how diverse the committership is etc

Reply:
@caniszczyk @bgrant0607 Updated to only list maintainers who have write access to etcd repository. Look forward to expanding more!